cbVideolink/js/cargobay.videolink.js

```diff
@@ -0,0 +1,91 @@
+/* ==========================================================================
+   Videolink
+
+   Initialize:
+   cargobay.videolink();
+
+   Support:
+   Latest Chrome
+   Latest FireFox
+   Latest Safari
+   IE10 and up
+   ========================================================================== */
+
+var cargobay = cargobay || {};
+
+cargobay.videolink = function(undefined) {
+
+    // Set global variables
+    var init;
+
+    // Main videolink function
+    _videolink = function() {
+
+      [].forEach.call(document.querySelectorAll('.js-videolink-play-link'), function(el) {
+
+        el.addEventListener('click', function(e) {
+
+          e.preventDefault();
+
+          // Get the data from the link to use on the iFrame and init variables
+          var provider = el.getAttribute('data-video-provider'),
+              id = el.getAttribute('data-video-id'),
+              videoContainer = el.parentNode.getElementsByClassName('js-videolink-container')[0],
+              template;
+
+
+          // Check what is the provider
+          switch(provider) {
+
+            case 'youtube' :
```

Comment:
Why repeat all of these 8 lines when only the first one changes. Maybe it's better to make a generator function to which you can pass the url of the streaming video?